core/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesRequest.java

```diff
@@ -56,16 +58,39 @@
 
     public FieldCapabilitiesRequest() {}
 
+    /**
+     * Returns <code>true</code> iff the results should be merged.
+     */
+    boolean isMergeResults() {
+        return mergeResults;
+    }
+
+    /**
+     * if set to <code>true</code> the response will contain only a merged view of the per index field capabilities. Otherwise only
+     * unmerged per index field capabilities are returned.
+     */
+    void setMergeResults(boolean mergeResults) {
+        this.mergeResults = mergeResults;
+    }
+
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         fields = in.readStringArray();
+        if (in.getVersion().onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED)) {
```

Comment:
yeah I will do this during the backport. I am afraid we will run into issues with the non-cross cluster tests that use field caps